File: OOP C++/lab2/Operands.cpp
#include "Operands.h"
#include <stdexcept>

int Operands::GetAndPop() {
    if(operands_.empty()){
        throw std::underflow_error("Stack underflow!");
    }
    int operand = operands_.top();
    operands_.pop();
    return operand;
}

int Operands::Top() {
    if(operands_.empty()){
        throw std::underflow_error("Stack underflow!");
    }
    return operands_.top();
}
void Operands::Push(int operand) {
    operands_.push(operand);
}

File: OOP Java/lab1/src/main/java/org/example/utility/Operands.java
package org.example.utility;

import java.util.Stack;

public class Operands {
    public int GetAndPop() {
        if(operands.empty()) {
            throw new RuntimeException("Incorrect operation that requires a number");
        }
        return operands.pop();
    }

    public int Top() {
        if(operands.empty()) {
            throw new RuntimeException("Incorrect operation that requires a number");
        }
        return operands.peek();
    }
    public void Push(Integer number) {
        operands.push(number);
    }

    public boolean IsEmpty() { return operands.empty(); }
    private final Stack<Integer> operands = new Stack<>();
}